R/Filter.R

```diff
@@ -619,6 +619,282 @@ makeFilter(
+    numeric.yes = unlist(lapply(data, function(x) class(x) == "numeric"))
+    df.num = data.frame(data[, numeric.yes])
+    interval = min(as.integer(nrow(data) / 3.0), 10L)
+    interval = max(interval, 2L)
+    cols.unique.len = unlist(lapply(df.num, function(x) length(unique(x))))
+    const.cols.yes = unlist(lapply(df.num, function(x) length(unique(x)) < 2L))
+    if (any(const.cols.yes)) {
+      df.num.uni =  lapply(df.num[const.cols.yes], as.factor)
+      df.num.cut = df.num[!const.cols.yes]
+      df.num.cut = data.frame(apply(df.num.cut, 2L, cut, interval))
+      df.num = cbind(df.num.cut, df.num.uni)
+    }
+    colnames(df.num) = colns[numeric.yes]
+    df.nonnum = data.frame(data[, !numeric.yes])
+    colnames(df.nonnum) =  colns[!numeric.yes]
+    data = cbind(df.nonnum, df.num)
+    return(data)
+}
+
+helper.cmi.praznik = function(criteria, preprocess = FALSE) {
+  candiates = c("JMI", "DISR", "JMIM", "MIM", "NJMIM", "MRMR", "CMIM")
+  checkmate::assert_choice(criteria, candiates)
+  criteria = paste0("praznik::", criteria)
+  function(task, nselect, ...) {
+    org.featnames = getTaskFeatureNames(task)
+    task = removeConstantFeatures(task)  # without removing constant features, praznik will generate Rcpp error
+    data = getTaskData(task)
+    featnames = getTaskFeatureNames(task)
+    targetname = getTaskTargetNames(task)
+    if (preprocess) data = preprocess.cmi.praznik(data)
+    X = data[, featnames]
+    Y = data[, targetname]
+    k = min(nselect, length(featnames))
+    k = max(k, 1)
+    input = list(X = X, Y = Y, k = k)
+    algo = eval(parse(text = criteria))
+    res = do.call(what = algo, args = input)
+    res$score
+    }}
+
+# fun must return a named vector of feature importance values. By convention the most important features receive the highest scores. If you are making use of the nselect option fun can either return a vector of nselect scores or a vector as long as the total numbers of features in the task filled with NAs for all features whose scores weren't calculated.
+# test consistency: d = generateFilterValuesData(iris.task, method = c("praznik.MIM", "anova.test"), nselect = 2)
+# test consistency: d = generateFilterValuesData(iris.task, method = c("praznik.MIM", "anova.test"), nselect = 4)
+
+#' Filters in the praznik package using mutual information criteria greedy search
+#' Features with higher scores are considered more important features
+#' Currently only tested with classification task with non-missing values
+#' candiates = c("JMI", "DISR", "JMIM", "MIM", "NJMIM", "MRMR", "CMIM")
+#' @rdname makeFilter
+#' @name makeFilter
+#'
+makeFilter(
+  name = "praznik.MIM",
+  desc = "conditional mutual information based feature selection filters",
+  pkg = "praznik",
+  supported.tasks = "classif",  # FIXME: still investigating if regression task could be used
+  supported.features = c("numerics", "factors", "integer", "character", "logical"),
+  fun = helper.cmi.praznik("MIM")
```

Comment:
Instead of `getFromNamespace` you could also just pass the function here, right?